src/components/Annotation.js

```diff
@@ -1,14 +1,30 @@
-import React from "react"
+// @flow
+import * as React from "react"
 import AnnotationLabel from "react-annotation/lib/Types/AnnotationLabel"
 
-import PropTypes from "prop-types"
+type Props = {
+  noteData: {
+    eventListeners: Object,
+    type: ?function,
+    screenCoordinates: Array<Array<number>>,
+    // What is this type supposed to be? It gets used only in a boolean context
+    // I mostly assume this is used to indicate the presence of `nx`, `ny`, `dx`, `dy`
+    coordinates: boolean,
+    nx: number,
+    ny: number,
+    dx: number,
+    dy: number,
+    // TODO: What should this be typed as?
+    note: Object,
+  }
+}
 
-class SemioticAnnotation extends React.Component {
+class SemioticAnnotation extends React.Component<Props, null> {
   render() {
     const { noteData } = this.props
     const { screenCoordinates } = noteData
 
-    noteData.type =
```

Comment:
Props are supposed to be immutable so I've switched this over to creating a new variable that gets used below.